balancer/balancer-subgraph-v2
# Fxpool oracle decimals

# Description

Adds a new property `fxOracleDecimals` to `Token` entity, and use this new property (instead of a hardcoded decimal) when computing for `latestFXPrice`.

### Motivation

We are updating support of FXPools in the  balancer-sor and our initial changes has been already merged on [balancer/balancer-sor#305](https://github.com/balancer/balancer-sor/pull/305). In this update, we are adding improvements on the accuracy of the returned values and fix the bug that was found by the sor development team (see https://github.com/balancer/balancer-sor/tree/fx-pool-debug).

`fxOracleDecimals` will primarily be used by `fxpoolMath` to ensure that all values are in wei from when we calculate things in the SOR. This will also include support of using our bb-usd oracles for the next version of our FxPool that we will be calling FxMetaPool.

## Type of change


## How should this be tested?

We've already deployed the changes on our subgraph clone here - https://thegraph.com/hosted-service/subgraph/xave-finance/balancer-v2

Simply select "fxOracleDecimals" from the dropdown and run it to see the results.

## Checklist:


### `dev` -> `master`
- [ ] I have [checked](https://balancer.github.io/balancer-subgraph-v2/status.html) that all beta deployments have synced
- [ ] I have [checked](https://api.thegraph.com/subgraphs/name/balancer-labs/balancer-polygon-prune-v2-beta/graphql?query=%0A%7B%0A++balancers%28block%3A%7Bnumber%3A1%7D%29%7B%0A++++id%0A++%7D%0A%7D) that the earliest block in the polygon pruned deployment is [block, date/time](https://polygonscan.com/block/block)
  - [ ] The earliest block is more than 24 hours old
- [ ] I have checked that core metrics are the same in the beta and production deployments

### Merges to `dev`
- [ ] I have checked that the graft base is not a pruned deployment
